src/geocel/detail/PolygonUtils.hh

```diff
@@ -0,0 +1,126 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file geocel/detail/LengthUnits.hh
+//! \brief NOTE: only use inside geocel; prefer celeritas/Units.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+//#define CELER_ICRT inline constexpr real_type
+#include <iostream>
+#include <vector>
+#include <corecel/cont/Array.hh>
+#include <corecel/math/ArrayUtils.hh>
+#include <orange/OrangeTypes.hh>
+
+namespace celeritas
+{
+namespace geoutils
+{
+
+template<typename T>
+using Real2 = Array<T, 2>;
+template<typename T>
+using VecReal2 = std::vector<Real2<T>>;
+template<typename T>
+using Real3 = Array<T, 3>;
+template<typename T>
+using VecReal3 = std::vector<Real3<T>>;
+
+enum PgonOrient : int
+{
+    clockwise = -1,
+    collinear = 0,
+    counterclockwise = 1,
+    kCW = clockwise,
+    kLine = collinear,
+    kCCW = counterclockwise
+};
+
+//---------------------------------------------------------------------------//
+// Find orientation of ordered triplet (a, b, c).
+// The function returns following values
+//  0 --> input points are collinear
+// -1 --> Clockwise
+// +1 --> Counterclockwise
+template<typename T>
+CELER_FUNCTION PgonOrient orientation_2D(Real2<T> const& a,
```

Comment:
There's no need to add the type, and the formatting is wrong anyway. I think this class can also explicitly use `real_type` rather than being templated. It also needs an `inline`.
```suggestion
CELER_FUNCTION Orientation orientation(Real2<T> const& a,
```